include/circt/Dialect/FIRRTL/FIRRTLExpressions.td

```diff
@@ -573,6 +573,18 @@ def AnalogInOutCastOp : FIRRTLOp<"analogInOutCast", [NoSideEffect]> {
   let verifier = "return ::verifyAnalogInOutCastOp(*this);";
 }
 
+// This operation converts from a struct to a bundle
+// type, or visa-versa.  FIRRTL source/destination types must be passive.
```

Comment:
Is it normal to have a type-sensitive conversion like this? It seems more natural to have a `bundleToStruct` and `structToBundle` which are then more restrictive in the types they take.

Reply:
It is semi-normal, there have been a couple disagreements on this point. In case you haven't seen it, there is [this discussion](https://llvm.discourse.group/t/open-meeting-1-14-dialect-conversion-and-type-conversion-the-question-of-cast-operations/2566/10) about adding a generic cast operation for any types to be used during specifically during dialect conversion.  It looks like this feature went in, its covered in the [tutorial](https://mlir.llvm.org/docs/Tutorials/Toy/Ch-4/).

Reply:
It is in the spirit of the other cast operations in firrtl for converting to and from rtl and std types.  Consistency seemed the most reasonable thing here.

Reply:
Sounds good to me. Thanks for the explanation. :+1: